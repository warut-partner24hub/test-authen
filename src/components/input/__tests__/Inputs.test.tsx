import { render, screen } from "@testing-library/react";
import Input from "../Inputs";
import { useForm } from "react-hook-form";

const Wrapper = ({ children }: any) => {
  const { register } = useForm();
  return children(register);
};

test("Input", () => {
  const { container } = render(
    <Wrapper>
      {(register: any) => (
        <Input
          name="first_name"
          label="First Name"
          type="text"
          placeholder="First Name"
          register={register}
        />
      )}
    </Wrapper>
  );
  expect(container).toMatchSnapshot();
});

describe("Input Component", () => {
  it("renders Input component correctly", () => {
    render(
      <Wrapper>
        {(register: any) => (
          <Input
            name="first_name"
            label="First Name"
            type="text"
            placeholder="First Name"
            register={register}
          />
        )}
      </Wrapper>
    );

    expect(screen.getByLabelText("First Name")).toBeInTheDocument();
  });

  it("shows error message", () => {
    render(
      <Wrapper>
        {(register: any) => (
          <Input
            name="first_name"
            label="First Name"
            type="text"
            placeholder="First Name"
            register={register}
            error="First name is required"
          />
        )}
      </Wrapper>
    );

    expect(screen.getByText("First name is required")).toBeInTheDocument();
  });

  it("disables the input", () => {
    render(
      <Wrapper>
        {(register: any) => (
          <Input
            name="first_name"
            label="First Name"
            type="text"
            placeholder="First Name"
            register={register}
            disable={true}
          />
        )}
      </Wrapper>
    );

    expect(screen.getByLabelText("First Name")).toBeDisabled();
  });
});
