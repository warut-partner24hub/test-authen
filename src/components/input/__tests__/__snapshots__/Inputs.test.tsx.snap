// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 1`] = `
<div>
  <div
    class="w-full"
  >
    <label
      class="block text-sm font-medium text-gray-700 mb-1"
      for="first_name"
    >
      First Name
    </label>
    <div
      class="relative"
    >
      <input
        class="block w-full px-4 py-2 border rounded border-gray-300 focus:outline-none focus:ring-2 focus:ring-blue-500 bg-white"
        id="first_name"
        name="first_name"
        placeholder="First Name"
        type="text"
      />
    </div>
  </div>
</div>
`;
